vaadin-testbench-core/src/main/java/com/vaadin/flow/component/grid/GridWrap.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.flow.component.grid;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import com.vaadin.flow.function.ValueProvider;
+import com.vaadin.testbench.unit.ComponentWrap;
+import com.vaadin.testbench.unit.MetaKeys;
+import com.vaadin.testbench.unit.Wraps;
+import com.vaadin.testbench.unit.component.GridKt;
+import com.vaadin.testbench.unit.internal.PrettyPrintTreeKt;
+
+@Wraps(Grid.class)
+public class GridWrap<T extends Grid<Y>, Y> extends ComponentWrap<T> {
+    /**
+     * Wrap grid for testing.
+     *
+     * @param component
+     *            target grid
+     */
+    public GridWrap(T component) {
+        super(component);
+    }
+
+    /**
+     * Get the amount of items in the grid.
+     *
+     * @return items in grid
+     */
+    public int size() {
+        return GridKt._size(getComponent());
+    }
+
+    /**
+     * Get the item at the given row index.
+     *
+     * @param row
+     *            row index of to get
+     * @return grid item on row
+     */
+    public Y getRow(int row) {
+        return GridKt._get(getComponent(), row);
+    }
+
+    /**
+     * Click on grid row.
+     *
+     * @param row
+     *            row to click
+     */
+    public void clickRow(int row) {
+        clickRow(row, 0);
+    }
+
+    /**
+     * Click on grid row with given button.
+     *
+     * @param row
+     *            row to click
+     * @see {@link com.vaadin.flow.component.ClickEvent#getButton()}
+     */
+    public void clickRow(int row, int button) {
```

Comment:
I know, this is technically correct, but this feels like no good API for end-user - wouldn't this be better with a enum `LEFT, RIGHT, MIDDLE` that maps to this keys?

Reply:
added/changed

Reply:
Thanks! Should there be a doubleClickRow method as well? This would match with the grid feature addDoubleClickListener where you typically open a dialog or navigate to the item's page.

Reply:
Perhaps. I'll look into it when I find out why dokka fails to build the javadoc